include/drivers/flash.h

```diff
@@ -79,6 +79,9 @@ __subsystem struct flash_driver_api {
 	flash_api_pages_layout page_layout;
 #endif /* CONFIG_FLASH_PAGE_LAYOUT */
 	const size_t write_block_size;
+#if defined(CONFIG_FLASH_ERASE_VALUE)
+	const u8_t erase_value;
```

Comment:
In theory it may not be possible to determine this value at compile time.  If we generalize QSPI to the point where it supports all flash serial memories that are JESD216 compatible we won't know whether the flash is NOR or NAND until runtime.

I think this should use a function rather than a constant to accommodate that possibility.

Reply:
As a follow-up: can somebody identify a device where the erase value is not 0xFF?  I'm not sure NOR vs NAND produces a difference.

Reply:
> 
> As a follow-up: can somebody identify a device where the erase value is not 0xFF? I'm not sure NOR vs NAND produces a difference.
> 

No I have failed to identify any FLASH/EEPROM devices that have erase values different from the 0xFF.

Reply:
> As a follow-up: can somebody identify a device where the erase value is not 0xFF? I'm not sure NOR vs NAND produces a difference.
> 
> ans: per #22063 "some stm32 embedded"

all STM32L0x and STM32L1x, and AFAIK Cypress PSOC 062/064.

Reply:
Thanks @utzig for the information, I have been able to go through some of the documents related to the MCUs (ST) you have mentioned. I have just skimmed the documentation, but what seems to be here the case is that the ST MCUs,  does not require erase and have its' flash  accessible via MCU address space the same way as "SRAM" (with proper register setup). So for them the erase_value would not have solved the problem. Some ST expert voice would be useful here.

Reply:
These STM32Lx NVMC performs flash erase operation as well. Most of SoC have access to embedded  flash via memory addresses (nRF as well).  So situuation is allmost the same as for nRF52, Erase value is just different,